FTheoryToolsInOSCAR.ipynb

```diff
@@ -860,32 +1124,1130 @@
+       "  1   0   0   0   0   0   0   0   0   0   0   0   0  -2   0   0   1   0   0\n",
+       "  0   0   0   1   0   0   0   0   0   0   0   0   0   0  -2   0   0   1   0\n",
+       "  0   0   2   0   0   0   0   0   0   0   0   0   0   0   0  -4   0   0   2\n",
+       "  0   1   0   0   0   0   0   0   0   0   0   0   0   1   0   0  -2   0   0\n",
+       "  0   1   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0  -2   0\n",
+       "  0   2   0   0   0   0   0   0   0   0   0   0   0   0   0   2   0   0  -4"
+      ]
+     },
+     "execution_count": 62,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "topological_intersection_numbers_among_nontrivial_ci_curves(qsm_model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "153a58b4-dc74-4a5b-b106-6da0da10548b",
+   "metadata": {},
+   "source": [
+    "The collection of the $C_{i}$ curves form a nodal curve. To every nodal curve one can associate a\n",
+    "(dual) graph. In this graph, every irreducible component of the nodal curve becomes a node/vertex\n",
+    "of the dual graph, and every nodal singularity of the nodal curve turns into an edge of the dual\n",
+    "graph. In the case at hand, this is rather simple.\n",
+    "\n",
+    "The $C_{i}$ curves turn into the irreducible components of the nodal curve. Certainly, we only need\n",
+    "to focus on the non-trivial $C_{i}$ curves. A non-trivial $C_{i}$ curve can split into multiple irreducible\n",
+    "components. This is taken into acccount when the nodes/vertices of the dual graph are constructed.\n",
+    "\n",
+    "The topological intersection numbers among the $C_{i}$ curves (or rather, their irreducible components)\n",
+    "tells us how many nodal singularities link the $C_{i}$ curves (or rather, their irreducible components)\n",
+    "in question. Hence, if the topological intersection numbers is zero, there is no edge between the\n",
+    "corresponding nodes. Otherwise, if the topological intersection number is positive, say $n$, then\n",
+    "there are exactly $n$ edges between the nodes in question."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "31f12218-d172-4eca-913b-c11c6a5e18d5",
+   "metadata": {},
+   "source": [
+    "The following method returns the dual graph of the QSM model in question.\n",
+    "Note that no labels are (currently) attached to the vertices/nodes or edges.\n",
+    "To understand/read this graph correctly, one of the following methods listed below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 63,
+   "id": "7a64a41b-38ee-4644-b307-016ecb32b32b",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Undirected graph with 21 nodes and the following edges:\n",
+       "(5, 1)(6, 5)(7, 6)(8, 7)(9, 4)(9, 8)(10, 1)(11, 4)(12, 3)(12, 10)(13, 3)(13, 11)(14, 1)(15, 4)(16, 3)(17, 3)(18, 2)(18, 14)(19, 2)(19, 15)(20, 2)(20, 16)(21, 2)(21, 17)"
```

Comment:
Not sure if we can add this to the notebook. But OSCAR has a function `visualize`. It will try to draw a picture of this graph. Does this look nice?

If not, can we use an extension to make it happen? For instance, the tutorial tests include the package Graphs.jl (as it is being used in https://nbviewer.org/github/thofma/HeckeTutorials.jl/blob/master/quadratic_forms.ipynb to draw pictures of graphs).

Maybe this could be nice to have? Maybe also for the simplified graph below?